packages/react-components/src/utils/common.ts

```diff
@@ -68,3 +74,94 @@ export const getIconFromUrl = (url: string): JSX.Element | undefined => {
   });
   return icon?.[1];
 };
+
+export type ResearchOutputState = {
+  title: string;
+  description: string;
+  link?: string;
+  type: string;
+  tags: readonly string[];
+  methods: string[];
+  organisms: string[];
+  environments: string[];
+  subtype?: string;
+  labCatalogNumber?: string;
+  teams: ComponentProps<typeof ResearchOutputContributorsCard>['teams'];
+  labs: ComponentProps<typeof ResearchOutputContributorsCard>['labs'];
+  authors: ComponentProps<typeof ResearchOutputContributorsCard>['authors'];
+  identifierType?: ResearchOutputIdentifierType;
+  identifier?: string;
+};
+
+export const isDirty = (
```

Comment:
I am not sure this should have been moved out of the component. This is def very unique to the Form rather than something that can be reused